Networking/Networking/Mapper/SubscriptionListMapper.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// Mapper: `Subscription` List
+///
+struct SubscriptionListMapper: Mapper {
+    /// Site we're parsing `Subscription`s for
+    /// We're injecting this field by copying it in after parsing responses, because `siteID` is not returned in any of the Subscription endpoints.
+    ///
+    let siteID: Int64
+
+    /// (Attempts) to convert a dictionary into `[Subscription]`.
+    ///
+    func map(response: Data) throws -> [Subscription] {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .formatted(DateFormatter.Defaults.dateTimeFormatter)
+        decoder.userInfo = [
+            .siteID: siteID
+        ]
+
+        do {
+            return try decoder.decode(SubscriptionListEnvelope.self, from: response).subscriptions
+        } catch {
+            return try decoder.decode([Subscription].self, from: response)
+        }
```

Comment:
FYI, this decoding strategy is [changing](https://github.com/woocommerce/woocommerce-ios/pull/9510). 
I think we could let Kiwi update this as they go through mappers but also does not feel nice to give them more work.

Maybe @itsmeichigo can share some guidelines in how to create mappers from now on?

Reply:
Thanks for the ping! When #9510 is merged, we should replace the `do`/`try` with the `hasDataEnvelope(in: response)` check. That should be sufficient!

Reply:
Thanks for the heads up! Updated in 5c7645c.